codec/src/main/java/io/netty/handler/codec/compression/Lz4FrameEncoder.java

```diff
@@ -163,6 +163,24 @@ private static int compressionLevel(int blockSize) {
     }
 
     @Override
+    protected ByteBuf allocateBuffer(ChannelHandlerContext ctx, ByteBuf msg, boolean preferDirect) {
+        int bufSize = 0;
+        int remaining = msg.readableBytes();
+        while (remaining > 0) {
+            int curSize = Math.min(blockSize, remaining);
+            remaining -= curSize;
+            // calculate the total compressed size of the current block (including header) and add to the total
+            bufSize += compressor.maxCompressedLength(curSize) + HEADER_LENGTH;
```

Comment:
@Scottmitch I am not sure if we really need a constructor in the base class or are you talking about `Lz4Encoder` here ?